warehouse/macaroons/services.py

```diff
@@ -156,6 +166,9 @@ def create_macaroon(
         user_id=None,
         oidc_publisher_id=None,
         additional=None,
+        # This is intended only for use in testing, to allow us to create legacy
+        # style Macaroons that do not have a permission scope.
+        _require_permission_scope=True,
```

Comment:
I don't necessarily think it's a better idea, but just to write it down: if we want to avoid this test-only state, we could probably have the relevant tests do explicit/raw DB inserts instead of calling `create_macaroon`.

Reply:
I'm little surprised we don't have a db-centric `MacaroonFactory` yet - so that might be a good addition to the testing stack, and we could create a [Trait](https://factoryboy.readthedocs.io/en/stable/reference.html#traits) to make the legacy-style if needed. Although I don't know how that might work with the need to have a serialized macaroon for testing, which we get from `create_macaroon()` - maybe we need a mock implementation of `IMacaroonService` to be used in tests?